lib-injection/src/test/shell/functions.sh

```diff
@@ -198,11 +213,110 @@ function deploy-app() {
-    sleep 5 && kubectl get pods
+    kubectl get pods
     echo "[Deploy] deploy-app done"
 }
 
+function deploy-app-auto() {
+    echo "[Deploy] deploy-app-auto: deploying app for library ${TEST_LIBRARY}"
+    deployment_name=test-${TEST_LIBRARY}-deployment
+    helm template lib-injection/common \
+      -f "lib-injection/build/docker/$TEST_LIBRARY/values-override.yaml" \
+      --set library="${TEST_LIBRARY}" \
+      --set as_deployment="true" \
+      --set deployment=${deployment_name} \
+      --set test_app_image="${LIBRARY_INJECTION_TEST_APP_IMAGE}" \
+       | kubectl apply -f -
+
+    echo "[Deploy] deploy-app-auto: waiting for deployments/${deployment_name} available"
+    kubectl rollout status deployments/${deployment_name} --timeout=5m
+    kubectl wait deployments/${deployment_name} --for condition=Available=True --timeout=5m
+    remove-terminating-pods
+    kubectl get pods
+
+    echo "[Deploy] deploy-app-auto: done"
+}
+
+function trigger-app-rolling-update() {
+    echo "[Deploy] trigger-app-rolling-update: updating deployment app ${TEST_LIBRARY}"
+    deployment_name=test-${TEST_LIBRARY}-deployment
+    kubectl set env deploy ${deployment_name} ENV_FOO=ENV_BAR
+
+    echo "[Deploy] trigger-app-rolling-update: waiting for deployments/${deployment_name} available"
+    kubectl rollout status deployments/${deployment_name} --timeout=5m
+    kubectl wait deployments/${deployment_name} --for condition=Available=True --timeout=5m
+    remove-terminating-pods
+    kubectl get pods
+
+    echo "[Deploy] trigger-app-rolling-update: done"
+}
+
+function check-for-env-vars() {
+    kubectl get pods
+    pod=$(latest-app-pod-auto)
+    echo "[Test] test for env vars ${pod}"
+    trace_sample_rate="0.90"
+    if [[ $CONFIG_NAME = "config-1" ]] ;  then
+        trace_sample_rate="0.50"
+    fi
+    # TODO: extend the list of tested env vars
+    kubectl get ${pod} -oyaml | grep -A1 DD_TRACE_SAMPLE_RATE | grep ${trace_sample_rate}
+}
+
+function check-for-pod-metadata() {
+    kubectl get pods
+    pod=$(latest-app-pod-auto)
+    echo "[Test] test for labels/annotations ${pod}"
+    [[ $TEST_LIBRARY = nodejs ]] && library=js || library=$TEST_LIBRARY
+    # TODO: check for label/annotation values not only the presence
+    enabled=$(kubectl get ${pod} -ojson | jq .metadata.labels | jq '."admission.datadoghq.com/enabled"')
+    if [[ $enabled != "\"true\"" ]]; then
+        echo "[Test] annotation 'admission.datadoghq.com/enabled' wasn't \"true\", got \"${enabled}\""
+        exit 1
+    fi
+    kubectl get ${pod} -ojson | jq .metadata.annotations | grep "admission.datadoghq.com/${library}-lib.version"
+    kubectl get ${pod} -ojson | jq .metadata.annotations | grep "admission.datadoghq.com/${library}-lib.config.v1"
+}
+
+function check-for-no-pod-metadata() {
+    kubectl get pods
+    pod=$(latest-app-pod-auto)
+    echo "[Test] test for labels/annotations ${pod}"
+    [[ $TEST_LIBRARY = nodejs ]] && library=js || library=$TEST_LIBRARY
```

Comment:
We are using this line many times. It is a "very special" code, perhaps we could put only on one place...